File: app/src/main/java/com/yanxing/base/BasePresenter.java
package com.yanxing.base;

/**
 * Created by lishuangxiang on 2016/9/13.
 */
public abstract class BasePresenter<T> {

    public T mView;

    /**
     * 销毁持有的View，看到一种说法：说不是任何情况下Activity的onDestory方法都会被调用，故对View进行弱引用
     */
    public void onDestroy() {
        this.mView = null;
    }

}

File: app/src/main/java/com/yanxing/base/BasePresenter.kt
package com.yanxing.base

/**
 * 基础Presenter，统一销毁持有的View
 * @author 李双祥 on 2018/7/5.
 */
open class BasePresenter<V : BaseView>{

    protected var mIView:V?=null

    /**
     * 绑定view
     */
    fun attachView(view:V){
        mIView=view
    }

    /**
     * 接触绑定
     */
    fun detachView() {
        mIView=null
    }
}